presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClientModule.java

```diff
@@ -39,14 +38,8 @@ protected void setup(Binder binder)
 
     private static void ensureCatalogIsEmpty(String connectionUrl)
     {
-        try {
-            Driver driver = new Driver();
-            Properties urlProperties = driver.parseURL(connectionUrl, null);
-            checkArgument(urlProperties != null, "Invalid JDBC URL for MySQL connector");
-            checkArgument(driver.database(urlProperties) == null, "Database (catalog) must not be specified in JDBC URL for MySQL connector");
-        }
-        catch (SQLException e) {
-            throw new RuntimeException(e);
-        }
+        checkArgument(isConnectionStringSupported(connectionUrl), "Invalid JDBC URL for MySQL connector");
+        ConnectionUrlParser parser = parseConnectionString(connectionUrl);
+        checkArgument(parser.getPath().isEmpty(), "Database (catalog) must not be specified in JDBC URL for MySQL connector");
```

Comment:
Looking at the code, `path` can be set to null. I think this should be `isNullOrEmpty(parser.getPath())`